packages/unimodules-face-detector-interface/android/src/main/java/org/unimodules/interfaces/facedetector/FaceDetector.java

```diff
@@ -1,14 +1,17 @@
 package org.unimodules.interfaces.facedetector;
 
-import android.os.Bundle;
+import android.net.Uri;
 
-import java.util.HashMap;
-import java.util.List;
+import java.io.IOException;
 import java.util.Map;
 
 public interface FaceDetector {
-  boolean isOperational();
-  List<Bundle> detectFaces(byte[] imageData, int width, int height, int rotation, int facing, double scaleX, double scaleY);
+
+  void detectFaces(Uri filePath, FacesDetectionCompleted listener, FaceDetectionError error) throws IOException;
+
+  boolean detectFaces(byte[] imageData, int width, int height, int rotation, boolean mirrored, double scaleX, double scaleY, FacesDetectionCompleted listener, FaceDetectionError error);
```

Comment:
I kind of wonder whether it's good that `detectFaces` returns two types of information:
- boolean — has the task been scheduled?
- callback — results/error

Do you think it makes sense to return `true`/`false` if we could just call the error callback?

Reply:
Partially agree :)

On the one hand, I agree with inconsistency it introduces. On the other, this returns false not when error occurs, but when detection is skipped due to user's settings.

Will change it by adding third callback (I know...) which informs about skipping. Is that okay?